Cars/CarsRepository/Repo/CarRepository.cs

```diff
@@ -0,0 +1,80 @@
+﻿using CarsRepository.Interface;
+using CarsData.Models;
+using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CarsRepository.Repo
+{
+    public class CarRepository : ICarRepository
+    {
+        private readonly AppDbContext appDbContext;
+
+        //EFDbContext context = new EFDbContext();
+        public CarRepository(AppDbContext appDbContext)
+        {
+            this.appDbContext = appDbContext;
+        }
+        public IEnumerable<Car> GetAllCars()
+        {
+            return appDbContext.Cars/*.Include(c => c.Categories)*/;
+        }
+
+        public IEnumerable<Car> GetFavoriteCars => appDbContext.Cars.Where(p => p.IsFavorite).Include(c => c.Categories);
+
+        public Car GetObjectCar(int CarId) => appDbContext.Cars.FirstOrDefault(p => p.CarId == CarId);
+
+        public IEnumerable<Car> GetCarsCategory(int Id) => appDbContext.Cars.Where(x => x.CategoryId == Id);
+        public IEnumerable<Car> GetCarsByCategory(int categoryId)
+        {
+            if(categoryId == -1)
```

Comment:
its betterto create another method to GetAllCars. it must be two different methods. 
Also read about IQueriable and IEnumerable and try to refactor your methods